src/desktop/wayland/popup/mod.rs

```diff
@@ -64,23 +71,31 @@ impl PopupKind {
     fn send_done(&self) {
         match *self {
             PopupKind::Xdg(ref t) => t.send_popup_done(),
+            // TODO nothing to do?
+            PopupKind::InputMethod(_) => {}
         }
     }
 
     fn location(&self) -> Point<i32, Logical> {
         let wl_surface = self.wl_surface();
 
-        with_states(wl_surface, |states| {
-            states
-                .data_map
-                .get::<XdgPopupSurfaceData>()
-                .unwrap()
-                .lock()
-                .unwrap()
-                .current
-                .geometry
-        })
-        .loc
+        match *self {
+            PopupKind::Xdg(_) => {
+                with_states(wl_surface, |states| {
+                    states
+                        .data_map
+                        .get::<XdgPopupSurfaceData>()
+                        .unwrap()
+                        .lock()
+                        .unwrap()
+                        .current
+                        .geometry
+                })
+                .loc
+            }
+            // TODO
+            PopupKind::InputMethod(ref t) => t.rectangle().loc.to_logical(1), // XXX
```

Comment:
I'd probably question the to_logical here with the scale of 1

Reply:
Yeah, this was a placeholder in my original PR, hence *both* `TODO` and `XXX` comments. :laughing:

I'm not sure exactly how it should be handled. It will need to be tested at a couple DPI scales.

Reply:
Should I get the scale from the compositor state using a trait?

Reply:
The scale depends on the output the popup is placed upon. Since you don't know the placement, you need to let the compositor provide it during rendering.

Reply:
I added a suggestion to solve this now, not sure if I can do it this way, but maybe it can be this simple?